File: w1_p5/3.py
import math
while True:
	people = input()
	money = []
	for i in range(int(people)):
		money.append(float(input()))
	if int(people) == 0:
		break
	average = sum(money) / len(money)
	
	move_money = []
	all_move = []
	for a in money:
			move_money.append(math.floor(100*(a - average)))
	for b in move_money:
		if (not b in all_move) and (b > 0):
			all_move.append(b/100)

	total = sum(all_move)
	print("$%.2f" %total)
	
	

File: w1_p5/3.c
#include <stdio.h>
#include <math.h>

int main()
{
	int N, i;
	double arr[1000];
	double sum, total;
	
	while(1){
		scanf("%d", &N);
		if(N == 0)
			break;
		sum=0, total=0;
		for(i=0; i<N; i++){
			scanf("%lf", &arr[i]);
			sum += arr[i];
		}
		sum /= N;
		sum = ceil(sum*100)/100;
		
		for(i=0; i<N; i++){
			if(arr[i] > sum)
				total += arr[i] - sum;
		}
		printf("$%.2f\n", total);
	}
	return 0;
}